ipmi-config/ipmi-config-category-core-lan6-conf-section.c

```diff
@@ -0,0 +1,843 @@
+          obj_cmd_rs = NULL;
+      }
+      if (ipv6_addresses_pos)
+          ipv6_addresses_str[ipv6_addresses_pos - 1] = '\0';
+
+      if (ipmi_config_section_update_keyvalue_output (state_data,
+                                                      kv,
+                                                      ipv6_addresses_str) < 0)
+        return (IPMI_CONFIG_ERR_FATAL_ERROR);
+    }
+
+  rv = IPMI_CONFIG_ERR_SUCCESS;
+ cleanup:
+  if (ipv6_addresses_str)
+      free(ipv6_addresses_str);
+  fiid_obj_destroy (obj_cmd_rs);
+  return (rv);
+}
+
+static ipmi_config_err_t
+ipv6_static_addresses_commit (ipmi_config_state_data_t *state_data,
+                              const char *section_name,
+                              const struct ipmi_config_keyvalue *kv)
+{
+  fiid_obj_t obj_cmd_rs = NULL;
+  uint8_t ipv6_address_bytes[16];
+  ipmi_config_err_t rv = IPMI_CONFIG_ERR_FATAL_ERROR;
+  ipmi_config_err_t ret;
+  uint8_t channel_number;
+
+  assert (state_data);
+  assert (section_name);
+  assert (kv);
+
+#if 0
+  /* XXX TODO lamont -- kv->value_input is a collection of IP addresess.  We need to:
+   * 1. break it up.
+   * 2. make sure that static_address_max is sufficent
+   * 3. iterate through the values setting them one at a time.
+   */
+  if (inet_pton (AF_INET6,
+                 kv->value_input,
+                 ipv6_address_bytes) != 1)
+    goto cleanup;
+
+  if (!(obj_cmd_rs = fiid_obj_create (tmpl_cmd_set_lan_configuration_parameters_ipv6_rs)))
+    {
+      pstdout_fprintf (state_data->pstate,
+                       stderr,
+                       "fiid_obj_create: %s\n",
+                       strerror (errno));
+      goto cleanup;
+    }
+
+  if ((ret = get_lan_channel_number (state_data,
+                                     section_name,
+                                     &channel_number)) != IPMI_CONFIG_ERR_SUCCESS)
+    {
+      rv = ret;
+      goto cleanup;
+    }
+
+  if (ipmi_cmd_set_lan_configuration_parameters_ipv6_static_addresses (state_data->ipmi_ctx,
+                                                                       channel_number,
+                                                                       ip_address_val,
+                                                                       obj_cmd_rs) < 0)
+    {
+      if (ipmi_config_param_errnum_is_non_fatal (state_data,
+                                                 obj_cmd_rs,
+                                                 &ret))
+        rv = ret;
+
+      if (rv == IPMI_CONFIG_ERR_FATAL_ERROR
+          || state_data->prog_data->args->common_args.debug)
+        pstdout_fprintf (state_data->pstate,
+                         stderr,
+                         "ipmi_cmd_set_lan_configuration_parameters_ip_address: %s\n",
+                         ipmi_ctx_errormsg (state_data->ipmi_ctx));
+
+      goto cleanup;
+    }
+
+  rv = IPMI_CONFIG_ERR_SUCCESS;
+ cleanup:
+  fiid_obj_destroy (obj_cmd_rs);
+#endif
```

Comment:
I left all the code in place, just added the flags, with a comment: "TODO: make this read-write."